[Bug] Unable to build on Mac M1 due to Homebrew's different default install path

**Describe the bug**
Since Homebrew installs itself onto `/opt/homebrew` on Mac M1, the build script fails when trying to build on Mac M1. Even after using VS Code to replace all instances of `/usr/local` with `/opt/homebrew`, the build still fails but a little further down the road.

**To Reproduce**
1. Follow prep instructions on README
- `brew install cmake freetype libvorbis sdl2 libpng jpeg libarchive libiconv`
- `git clone https://github.com/Drewol/unnamed-sdvx-clone`
- `cd unnamed-sdvx-clone`
- `git submodule update --init --recursive`
2. Use VS Code to replace all instances of `/usr/local` in the project folder with `/opt/homebrew`.
2. Attempt to build the project
- `./mac-cmake.sh`
- `make`

**Expected behavior**
An executable appears in the bin folder.

**`make` log (before replacing `/usr/local`)**
```
...
[ 38%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Config.cpp.o
[ 39%] Building CXX object Shared/CMakeFiles/Shared.dir/src/ConfigEntry.cpp.o
[ 39%] Building CXX object Shared/CMakeFiles/Shared.dir/src/FileStream.cpp.o
[ 40%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Interpolation.cpp.o
[ 40%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Jobs.cpp.o
[ 41%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Log.cpp.o
/Users/alex/git/unnamed-sdvx-clone/Shared/src/Log.cpp:145:9: warning: enumeration value '_Length' not handled in switch [-Wswitch]
        switch(severity)
               ^
/Users/alex/git/unnamed-sdvx-clone/Shared/src/Log.cpp:145:9: note: add missing switch cases
        switch(severity)
               ^
1 warning generated.
[ 41%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Math.cpp.o
[ 42%] Building CXX object Shared/CMakeFiles/Shared.dir/src/MemoryStream.cpp.o
[ 42%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Path.cpp.o
[ 43%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Random.cpp.o
[ 43%] Building CXX object Shared/CMakeFiles/Shared.dir/src/String.cpp.o
[ 43%] Building CXX object Shared/CMakeFiles/Shared.dir/src/StringEncodingConverter.cpp.o
/Users/alex/git/unnamed-sdvx-clone/Shared/src/StringEncodingConverter.cpp:9:10: fatal error: 'archive_entry.h' file not found
#include "archive_entry.h"
         ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [Shared/CMakeFiles/Shared.dir/src/StringEncodingConverter.cpp.o] Error 1
make[1]: *** [Shared/CMakeFiles/Shared.dir/all] Error 2
make: *** [all] Error 2
```

**`make` log (after replacing `/usr/local`)**
```
...
[ 45%] Building CXX object Shared/CMakeFiles/Shared.dir/src/TextStream.cpp.o
[ 45%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Transform.cpp.o
[ 46%] Building CXX object Shared/CMakeFiles/Shared.dir/src/UTF8.cpp.o
[ 46%] Building CXX object Shared/CMakeFiles/Shared.dir/src/VectorMath.cpp.o
[ 47%] Building CXX object Shared/CMakeFiles/Shared.dir/src/MacOS/Thread.cpp.o
[ 47%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Unix/Debug.cpp.o
[ 48%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Unix/File.cpp.o
[ 48%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Unix/Files.cpp.o
[ 49%] Building CXX object Shared/CMakeFiles/Shared.dir/src/Unix/Path.cpp.o
/Users/alex/git/unnamed-sdvx-clone/Shared/src/Unix/Path.cpp:104:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
1 warning generated.
[ 49%] Building CXX object Shared/CMakeFiles/Shared.dir/stdafx.cpp.o
[ 50%] Linking CXX static library ../lib/libShared_Release.a
[ 50%] Built target Shared
[ 50%] Building CXX object Graphics/CMakeFiles/Graphics.dir/src/Font.cpp.o
[ 51%] Building CXX object Graphics/CMakeFiles/Graphics.dir/src/Gamepad_Impl.cpp.o
[ 51%] Building CXX object Graphics/CMakeFiles/Graphics.dir/src/Image.cpp.o
/Users/alex/git/unnamed-sdvx-clone/Graphics/src/Image.cpp:8:10: fatal error: 'libpng16/png.h' file not found
#include "libpng16/png.h"
         ^~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [Graphics/CMakeFiles/Graphics.dir/src/Image.cpp.o] Error 1
make[1]: *** [Graphics/CMakeFiles/Graphics.dir/all] Error 2
make: *** [all] Error 2
```

**System**
MacBook Air 2020
M1 Apple Silicon
256 GB SSD
8 GB RAM

I've decided to create symlinks `/usr/local/include` and `/usr/local/opt` to point to the `/opt/homebrew/` counterparts, which seems to have helpd quite a bit. However, now I'm getting an error at this stage:

```
[ 71%] Building CXX object Main/CMakeFiles/usc-game.dir/src/CalibrationScreen.cpp.o
In file included from /Users/alex/git/unnamed-sdvx-clone/Main/src/CalibrationScreen.cpp:6:
/Users/alex/git/unnamed-sdvx-clone/Main/../third_party/nuklear/nuklear.h:422:1: error: '_dummy_array422' declared as an array with a negative size
NK_STATIC_ASSERT(sizeof(nk_size) >= sizeof(void*));
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/alex/git/unnamed-sdvx-clone/Main/../third_party/nuklear/nuklear.h:291:75: note: expanded from macro 'NK_STATIC_ASSERT'
  #define NK_STATIC_ASSERT(exp) typedef char NK_UNIQUE_NAME(_dummy_array)[(exp)?1:-1]
                                                                          ^~~~~~~~~~
/Users/alex/git/unnamed-sdvx-clone/Main/../third_party/nuklear/nuklear.h:423:1: error: '_dummy_array423' declared as an array with a negative size
NK_STATIC_ASSERT(sizeof(nk_ptr) >= sizeof(void*));
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/alex/git/unnamed-sdvx-clone/Main/../third_party/nuklear/nuklear.h:291:75: note: expanded from macro 'NK_STATIC_ASSERT'
  #define NK_STATIC_ASSERT(exp) typedef char NK_UNIQUE_NAME(_dummy_array)[(exp)?1:-1]
                                                                          ^~~~~~~~~~
2 errors generated.
make[2]: *** [Main/CMakeFiles/usc-game.dir/src/CalibrationScreen.cpp.o] Error 1
make[1]: *** [Main/CMakeFiles/usc-game.dir/all] Error 2
make: *** [all] Error 2
```

@Drewol 
Attempting to compile on my intel mac yields the same error. The output from `mac-cmake.sh` seems to indicate that CMake is finding the wrong version of `libpng`:
```-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Freetype: /usr/local/lib/libfreetype.dylib (found version "2.11.0")
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/libz.tbd (found version "1.2.11")
-- Found SDL2: /Library/Frameworks/SDL2.framework
-- Found PNG: /usr/local/lib/libpng.dylib (found version "1.4.12")
-- Found JPEG: /usr/local/lib/libjpeg.dylib (found version "80")
-- Found Vorbis: /usr/local/include
-- Found OGG: optimized;/usr/local/lib/libogg.dylib;debug;/usr/local/lib/libogg.dylib
-- Found LibArchive: /usr/local/opt/libarchive/lib/libarchive.dylib (found version "3.5.1")
-- Performing Test Iconv_IS_BUILT_IN
-- Performing Test Iconv_IS_BUILT_IN - Failed
-- Found Iconv: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/libiconv.tbd (found version "1.11")
Enabling crash dumps
CMake Error at third_party/CMakeLists.txt:20 (add_subdirectory):
  The source directory

    /Users/matthew/My Applications/unnamed-sdvx-clone/third_party/cpr

  does not contain a CMakeLists.txt file.


CMake Error at third_party/CMakeLists.txt:21 (target_include_directories):
  Cannot specify include directories for target "cpr" which is not built by
  this project.


CMake Error at third_party/CMakeLists.txt:25 (add_subdirectory):
  The source directory

    /Users/matthew/My Applications/unnamed-sdvx-clone/third_party/discord-rpc

  does not contain a CMakeLists.txt file.


CMake Error at third_party/CMakeLists.txt:26 (target_include_directories):
  Cannot specify include directories for target "discord-rpc" which is not
  built by this project.


-- Found Git: /usr/bin/git (found version "2.30.1 (Apple Git-130)")
-- Configuring incomplete, errors occurred!
See also "/Users/matthew/My Applications/unnamed-sdvx-clone/CMakeFiles/CMakeOutput.log".
See also "/Users/matthew/My Applications/unnamed-sdvx-clone/CMakeFiles/CMakeError.log".```
How could I tell CMake to use the correct version?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -94,12 +94,11 @@ jobs:
             - name: brew update
               run: brew update
             - name: Install packages
-              run: brew install freetype libvorbis sdl2 libpng jpeg libarchive
+              run: brew install freetype libvorbis sdl2 libpng jpeg-turbo libarchive
             - name: cmake
-              run: |
-                eval "$(brew shellenv)"
-                rm ./vcpkg.json
-                cmake . -DLibArchive_LIBRARY=$HOMEBREW_PREFIX/opt/libarchive/lib/libarchive.dylib -DLibArchive_INCLUDE_DIR=$HOMEBREW_PREFIX/opt/libarchive/include -DCMAKE_BUILD_TYPE=Release
+              env:
+                DCMAKE_BUILD_TYPE: Release
+              run: cmake .
             - name: make
               run: make
             - name: Upload artifact

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,6 +42,29 @@ endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
 
 set(CMAKE_MACOSX_RPATH 1)
 
+# Set library paths for MacOS
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(MACOSX TRUE)
+
+    # Set homebrew's include dir
+    execute_process(
+        COMMAND brew --prefix
+        OUTPUT_VARIABLE HOMEBREW_PREFIX
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        COMMAND_ERROR_IS_FATAL ANY
+    )
+    include_directories("${HOMEBREW_PREFIX}/include")
+
+    # Libarchive is shipped as a keg so we must get its path manually
+    execute_process(
+        COMMAND brew --prefix libarchive
+        OUTPUT_VARIABLE LIBARCHIVE_PREFIX
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        COMMAND_ERROR_IS_FATAL ANY
+    )
+    set(LibArchive_INCLUDE_DIR "${LIBARCHIVE_PREFIX}/include")
+endif()
+
 # Set folder where to find FindXXX.cmake and
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Modules/")
 

--- a/README.md
+++ b/README.md
@@ -97,8 +97,8 @@ To run from Visual Studio, go to Properties for Main > Debugging > Working Direc
 ### macOS:
 0. Clone the project using `git` and then run `git submodule update --init --recursive` to download the required submodules.
 1. Install dependencies
-	* [Homebrew](https://github.com/Homebrew/brew): `brew install cmake freetype libvorbis sdl2 libpng jpeg libarchive libiconv`
-2. Run `mac-cmake.sh` and then `make` from the root of the project.
+	* [Homebrew](https://github.com/Homebrew/brew): `brew install cmake freetype libvorbis sdl2 libpng jpeg-turbo libarchive libiconv`
+2. Run `cmake -DCMAKE_BUILD_TYPE=Release .` and then `make` from the root of the project.
 3. Run the executable made in the 'bin' folder.
 
 ### Embedded (Raspberry Pi):

--- a/mac-cmake.sh
+++ b/mac-cmake.sh
@@ -1,2 +1,0 @@
-eval "$(brew shellenv)"
-cmake . -DLibArchive_LIBRARY=$HOMEBREW_PREFIX/opt/libarchive/lib/libarchive.dylib -DLibArchive_INCLUDE_DIR=$HOMEBREW_PREFIX/opt/libarchive/include -DCMAKE_BUILD_TYPE=Release

